src/Grpc.Net.ClientFactory/GrpcClientFactory.cs

```diff
@@ -32,6 +36,25 @@ public abstract class GrpcClientFactory
         /// <typeparam name="TClient">The gRPC client type.</typeparam>
         /// <param name="name">The configuration name.</param>
         /// <returns>A gRPC client instance.</returns>
-        public abstract TClient CreateClient<TClient>(string name) where TClient : ClientBase;
+        public abstract TClient CreateClient<TClient>(string name) where TClient : class;
+
+        /// <summary>
+        /// Create a <see cref="CallInvoker"/> instance for the specified configuration name.
+        /// </summary>
+        /// <param name="serviceProvider">The service provider.</param>
+        /// <param name="name">The configuration name.</param>
+        /// <returns>A gRPC call invoker instance.</returns>
+        protected static CallInvoker CreateCallInvoker(IServiceProvider serviceProvider, string name)
```

Comment:
just to note: I was on the fence here as to whether this might prefer to be `protected virtual`, to allow `override`; not hugely invested either way, but probably merits a second look